dbms/src/Storages/DeltaMerge/File/DMFileReader.cpp

```diff
@@ -528,37 +488,37 @@ ColumnPtr DMFileReader::readColumn(const ColumnDefine & cd, size_t start_pack_id
                   return readFromDiskOrSharingCache(cd, type_on_disk, 0, dmfile->getPacks(), dmfile->getRows());
               });
 
-        auto column = type_on_disk->createColumn();
-        column->insertRangeFrom(*column_all_data, next_row_offset - read_rows, read_rows);
-        return convertColumnByColumnDefineIfNeed(type_on_disk, std::move(column), cd);
+        auto mut_column = type_on_disk->createColumn();
+        mut_column->insertRangeFrom(*column_all_data, pack_offset[start_pack_id], read_rows);
+        column = std::move(mut_column);
```

Comment:
Why not directly `return convertColumnByColumnDefineIfNeed(type_on_disk, std::move(column), cd);` in this line?